src/app/doc_observer.h

```diff
@@ -97,8 +97,10 @@ namespace app {
     // The collapsed/expanded flag of a specific layer changed.
     virtual void onLayerCollapsedChanged(DocEvent& ev) { }
 
-    // The visibility flag of a specific layer is going to change/changed.
-    virtual void onBeforeLayerVisibilityChange(DocEvent& ev, bool newState) { }
+    // The 'visibility' or 'editable' flag of a specific layer is going
+    // to change/changed.
+    virtual void onBeforeLayerStatusChange(DocEvent& ev, bool newState) { }
```

Comment:
I don't like changing the name of this function just because when the lock switches it has to behave the same as when changing visibility. Also there is a onAfterLayerVisibilityChange that now wouldn't have the onBefore.
So, I see two alternatives:
1. Keep the new name, change the onAfterLayerVisibilityChange by onAfterLayerStatusChange for consistency. And add something (a new parameter maybe?) to let the implementer know if the status change is about the visibility or the editable flag.
2. Revert this change to preserve onBeforeLayerVisibilityChange as it was. Add a new onBeforeLayerEditableChange and use it.

By the way, I like more the second approach, but that's just my opinion.